src/components/Alert/Alert.tsx

```diff
@@ -0,0 +1,86 @@
+import { useMemo, useRef, useState } from "react";
+
+import AlertCard from "./AlertCard";
+import useAlarm from "./hooks/useAlarm";
+
+import {
+  Active as ActiveAlarmIcon,
+  Close,
+  Inactive as InactiveAlarmIcon,
+} from "@/assets/icon";
+import useIntersection from "@/hooks/useIntersection";
+import useOutsideClick from "@/hooks/useOutsideClick";
+import { useUserStore } from "@/hooks/useUserStore";
+
+function Alert() {
+  const user = useUserStore((state) => state.user);
+  const [showDropdown, setShowDropdown] = useState(false);
+  const { isLoading, hasNext, refetch, alerts, totalCount } = useAlarm({
+    userId: user!.id,
+  });
```

Comment:
```suggestion
if (!user?.id) return null;
const { isLoading, hasNext, refetch, alerts, totalCount } = useAlarm({ userId: user.id });
```

이렇게 수정하는 게 좀 더 안전한 코드라고 생각합니다 🤓

Reply:
아예 부모로부터 `userId`를 전달 받도록 수정했습니다 😆

```tsx
// Header.tsx
{user && <Alert userId={user.id} />}
```
```tsx
interface AlertProps {
  userId: string;
}

function Alert({ userId }: AlertProps) {
  const [showDropdown, setShowDropdown] = useState(false);
  const { isLoading, hasNext, refetch, alerts, totalCount } = useAlarm({
    userId,
  });
...
```